File: problem/Min_Absolute_Difference/minimumAbsDifference.js
/**
 * @param {number[]} arr
 * @return {number[][]}
 */
var minimumAbsDifference = function(arr) {
    arr.sort((a, b) => a - b);
    let minDiff = Infinity;
    let res = [];

    for (let i = 1; i < arr.length; i++) {
        let diff = arr[i] - arr[i - 1];
        if (diff < minDiff) {
            minDiff = diff;
            res = [[arr[i - 1], arr[i]]];
        } else if (diff === minDiff) {
            res.push([arr[i - 1], arr[i]]);
        }
    }
    return res;
};

File: problem/Min_Absolute_Difference/minimumAbsDifference.py
class Solution:
    def minimumAbsDifference(self, arr):
        arr.sort()
        min_diff = float('inf')
        res = []
        
        for i in range(1, len(arr)):
            diff = arr[i] - arr[i - 1]
            if diff < min_diff:
                min_diff = diff
                res = [[arr[i - 1], arr[i]]]
            elif diff == min_diff:
                res.append([arr[i - 1], arr[i]])
                
        return res